src/app/auth/commandAuth.ts

```diff
@@ -105,7 +105,9 @@ const publish = async (request: PublishRequest): Promise<void> => {
         _id: new mongo.ObjectId(request.bountyId),
     });
 
-    if (dbBountyResult.gate && !(await DiscordUtils.hasAllowListedRole(request.userId, request.guildId, dbBountyResult.gate))) {
+    const gate = (dbBountyResult.gate || (dbBountyResult.gateTo && dbBountyResult.gateTo.map( g => g.discordId)));
```

Comment:
Could you document falsy behavior on arrays?

as well, might consider refactoring `g` to `roleObject`

Reply:
I'm not sure this will work? 

If I'm understanding correctly:

```
console.log([
    {
        discordId: 1
    }, {
        discordId: 1
    }, {
    }, {
        discordId: 1
    }
].map(r => r.discordId))
console.log(!![ 1, 1, undefined, 1 ]);
console.log(true && [
    {
        discordId: 1
    }, {
        discordId: 1
    }, {
    }, {
        discordId: 1
    }
].map(r => r.discordId))
```

produces:

```
[ 1, 1, undefined, 1 ]
true
[ 1, 1, undefined, 1 ]
```

Reply:
This is what I was shooting for:

If .gate is set, return that (it will be an array with a single value: [discordId])

Otherwise, if .gateTo exists, return the discordIds from the contained objects ([discordId, discordId, discordId]). There *should* always be a discord Id in each object.

Right now .gateTo will also only have one role in it, similar to .gate, but in the future we will want to support multiple.

Reply:
Put it in the code comments